src/mdnsresponder.c

```diff
@@ -427,18 +427,18 @@ static ETSTimer mdns_announce_timer;
 void mdns_clear() {
     sdk_os_timer_disarm(&mdns_announce_timer);
     
-    xSemaphoreTake(gDictMutex, portMAX_DELAY);
-
-    mdns_rsrc *rsrc = gDictP;
-    gDictP = NULL;
+    if(xSemaphoreTake(gDictMutex, portMAX_DELAY)) {
```

Comment:
Are these guard conditions necessary? Since it uses portMAX_DELAY, xSemaphoreTake will only return when it had successfully acquired semaphore.

Reply:
It is necessary when INCLUDE_vTaskSuspend is set to 0. Default value is 1, but someone could use 0 for custom projects.